content/inside-rust/stage0-redesign/index.md

```diff
@@ -0,0 +1,116 @@
+
+1. [Explain how the current stage 0 bootstrapping sequence works](#current-model), and
+2. [Explain how the new stage 0 bootstrapping sequence works after the redesign](#new-model), and
+3. [Discuss why the redesigned stage 0 bootstrapping sequence is more preferable](#better).
+
+# The current stage 0 bootstrap sequence {#current-model}
+
+![Current stage 0 bootstrap sequence](./stage0-current.svg)
+
+Currently, [bootstrap] downloads a pre-built beta rustc as the initial compiler (stage 0 rustc).
+
+- To produce a stage 1 rustc, we need to produce a stage 0 std. This stage 0 std is built from in-tree standard library sources with `cfg(bootstrap)` active.
+- To produce a stage 2 rustc, we need to produce a stage 1 std. However, this is where things get weird. The stage 1 std is built from the *same* in-tree standard library sources (but with `cfg(bootstrap)` inactive). This is what we meant by "the standard library has to support being buildable by two compiler versions."
+
+# The redesigned stage 0 bootstrap sequence {#new-model}
+
+![Redesigned stage 0 bootstrap sequence](./stage0-next.svg)
+
+In the [redesigned stage 0 bootstrapping sequence][stage0-redesign-pr] we instead download *both* the pre-built beta rustc as the stage 0 compiler, and the pre-built beta std as the stage 0 std, *instead* of building stage 0 std from in-tree sources.
+
+- When producing a stage 1 rustc, we already have the stage 0 std (as it is the pre-built std).
+- When producing a stage 2 rustc, the stage 1 std is then the std built from in-tree std sources with the stage 1 rustc.
+
+# Why is the redesigned stage 0 bootstrap sequence better? {#better}
+
+There are several benefits of the redesigned stage 0 bootstrap sequence:
+
+1. We no longer have to use `cfg(bootstrap)` in the standard library sources for intrinsics and lang items to distinguish when being built by the beta rustc vs the in-tree rustc, because the standard library now only has to be buildable by exactly one compiler version (the current stage rustc).
+2. It significantly reduces cognitive complexity, as the redesigned bootstrap sequence is much more coherent and aligns better with how contributors expect the staging to work. We no longer have a "strange" setup where the stage 1 compiler was built from a *beta* rustc with an *in-tree* std. Now, the stage 1 compiler is built from a beta rustc and a beta std.
+
+# In terms of bootstrap invocations and bootstrap config, what does this redesign mean?
+
+The minimum stage to check, build and test the standard library is now stage 1. `./x {check,build,test} library --stage=0` are now no-ops.
+
+For `profile = "library"` users, like aforementioned, the default check, build and test stage are now bumped to 1. `download-rustc = "if-unchanged"` is enabled by default so a pre-built CI rustc is used to help you avoid needing to rebuild the compiler while working on the standard library if there are no compiler changes.
```

Comment:
i find this very hard to parse given the comma clause earlier.
```suggestion
For `profile = "library"` users, the default check, build, and test stage are now bumped to 1. `download-rustc = "if-unchanged"` is enabled by default, which downloads a pre-built CI rustc instead of building the compiler if there are no compiler changes, allowing you to build the standard library without building the compiler.
```

Reply:
Reworded (round 5).